service/src/main/java/com/bankbazaar/kafka/service/producer/KafkaProducer.java

```diff
@@ -1,25 +1,41 @@
 package com.bankbazaar.kafka.service.producer;
 
 import com.bankbazaar.kafka.core.model.Data;
+import com.bankbazaar.kafka.dto.model.FileStatusDto;
 import com.bankbazaar.kafka.service.mapper.DataMapper;
 import com.bankbazaar.kafka.dto.model.DataDto;
-import lombok.NoArgsConstructor;
+import com.bankbazaar.kafka.service.model.Response;
+import com.bankbazaar.kafka.service.service.FileStatusService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.kafka.core.KafkaTemplate;
 import org.springframework.stereotype.Service;
 
+import java.util.Date;
+
 @Service
 public class KafkaProducer {
-    private  static  final  String TOPIC = "create_CSV";
+    private  static  final  String TOPIC = "Create_CSV";
 
     @Autowired
     private  KafkaTemplate<String, Data> kafkaTemplate;
+    @Autowired
+    private FileStatusService fileStatusService;
 
     @Autowired
     private DataMapper mapper;
 
-    public void sendData(DataDto data)
+    public Response sendData(DataDto data)
     {
+        FileStatusDto fileData = new FileStatusDto();
+        fileData.setFileName(data.getFileName());
+        fileData.setStatus("NEW");
+        FileStatusDto response = fileStatusService.insert(fileData);
```

Comment:
This method should be like this
1. get Data object
2. Pass this object to the FileStatusService
3. FileStatusService should add the new entry and send back the entity
4. Push the request to the Kafka producer for processing
5. Create Response from the entity returned in step #3 and send it back

Reply:
This has been implemented